### ISSUE TITLE
HTML rendering broken whe allOf and oneOf

### ISSUE THREAD

**Issue by LasneF** [Redocly/redocly-cli#1594]
**Describe the bug**

given a specification with no error , no warning using redocly lint , the documentation generated via redocly build-docs "crashes" at rendering , and make it unusable

**To Reproduce**
given the Open API description attached 

run ``redocly build-docs .\component-descriptor.yml `` 

Open the HTML File produce , 

select for kind  the value CAT , the HTML rendering is ok 
select for kind the value DOG ; the HTML rendering is broken and does not allow anymore to choose a value 

**Expected behavior**

Either the API specification is considered as invalid are not renderable , and the build doc should shoot an error ,can be at the build doc or at the linter level 

Either it is considered as  correct and the rendering should be OK 

**Logs**
NA 

**OpenAPI description**
cf attached file (notice that yml or json does not matter) 

**Redocly Version(s)**
redocly --version                                                                                                                        
1.16.0

**`Node.js` Version(s)**
node --version  
v18.16.0

**Additional context**

[component-descriptor.json](https://github.com/user-attachments/files/15902946/component-descriptor.json)

**Comment by tatomyr** [comment]
If you remove the oneOf inside Dog, it starts working. On one hand, it's redundant, and removing it even makes the documentation a bit more readable. On the other hand, the schema is valid, so I'd expect Redoc to render it correctly.

@Redocly/keyboard-warriors, could you check if it's a bug on Redoc's end?

**Comment by LasneF** [comment]
@tatomyr  you are right , the sample has been over simplified , as the real use case was to have 
AllOf  (  something + oneOf ( A or B) + oneOf(C+D) )
